File: chp01_Singleton/Kotlin/DatabaseConnection_basic.kt
object DatabaseConnection { 

    private var instance: DatabaseConnection? = null 

    init { 

        // Initialize the database connection here 

        println("Database connection initialized") 

    }

    fun getInstance(): DatabaseConnection { 

        if (instance == null) { 

            instance = DatabaseConnection 

        } 

        return instance!! 

    } 
     

    fun executeQuery(query: String): String { 

        // Execute the query and return the result 

        println("Executing query: $query") 

        return "Query executed successfully" 

    } 

}  

fun main() {  

    val dbInstance = DatabaseConnection.getInstance() 

val result = dbInstance.executeQuery("SELECT * FROM users") 

println(result)  

} 

File: chp01_Singleton/Swift/DatabaseConnection_basic.swift
import Foundation

class DatabaseConnection {
    static let sharedInstance = DatabaseConnection()
    
    private init() {}
    
    func connectToDatabase() {
        // Connect to the database here
        print("Connected to database")
    }
    
    func executeQuery(query: String) {
        // Execute the query on the database
        print("Executing query: \(query)")
    }
}

// Usage
let dbConnection = DatabaseConnection.sharedInstance
dbConnection.connectToDatabase()
dbConnection.executeQuery(query: "SELECT * FROM users")